### PR TITLE
Adding status after extract command

### PR DESCRIPTION
**Pull request by Nyholm**
Sorry for a messy PR. This PR does also refactor `NoteParser` and`MetadataWriter`.

NoteParser is needed to get structured data from a XML note. The MetadataWriter writes metadata (like notes) to the Catalogue. It makes sure we do not have duplicates. 

The `CatalogueCounter` could be used for #52 

screen shot 2017-08-13 at 17 25 58

### CODE REVIEW

File: Catalogue/MetadataWriter.php [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+ * (c) PHP Translation team <tobias.nyholm@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Translation\Bundle\Catalogue;
+
+use Symfony\Component\Translation\MessageCatalogue;
+
```
Comment (Nyholm) [inline Catalogue/MetadataWriter.php]: Add class comment

File: Catalogue/NoteParser.php [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Translation\Bundle\Catalogue;
+
+use Symfony\Component\Translation\MetadataAwareInterface;
+
+class NoteParser
```
Comment (Nyholm) [inline Catalogue/NoteParser.php]: Add class comment

**Comment by rvanlaak** [comment]
I've completely missed this PR, but think it's a great addition.

Only one though; wouldn't it be more clear to store `errors` inside of the `Importer` instead of passing it by reference.

So, `$this->errors[$key] = $collection->getErrors()` and adding `$importer->getErrors()`, instead of `&$errors = $collection->getErrors()`. That way the signature of the method also doesn't have to change.

**Comment by Nyholm** [comment]
That would make the Importer service stateful. Hm. The best thing would be if we could change the return value of the importer to make it return a `ImportResult`. Then do `$importResult->getMessageCatalogues()` and `$importResult->getErrors()`.

**Comment by rvanlaak** [comment]
You're right that the importer service should have state. Given the number of parameters, and possible future features, transforming the output to a value object would be the best option.

**Comment by Nyholm** [comment]
I’ll make some updates today.

Med vänliga hälsningar
Tobias Nyholm

**Pull request merged**